### PR TITLE
bridge: Ensure our host handling bridge has priority

### PR DESCRIPTION
**Pull request by petervo**

### CODE REVIEW

File: src/bridge/stub.c [hunk 1]
```diff
@@ -220,7 +220,7 @@ print_rules (void)
   GList *bridges = NULL;
   CockpitTransport *transport = cockpit_interact_transport_new (0, 1, "--");
 
-  router = setup_router (NULL, &bridges);
+  router = setup_router (transport, &bridges);
```
Comment (martinpitt) [inline src/bridge/stub.c]: How is that related to the above list reverse / priority change? It looks like this rather belongs into the previous commit?

File: src/bridge/stub.c [hunk 2]
```diff
@@ -202,6 +214,24 @@ run_bridge (const gchar *interactive)
...
+
+  router = setup_router (NULL, &bridges);
+
+  cockpit_router_dump_rules (router);
+
+  g_object_unref (router);
+  g_list_free (bridges);
+  g_object_unref (transport);
+  cockpit_packages_free (packages);
+  packages = NULL;
```
Comment (martinpitt) [inline src/bridge/stub.c]: Same issue with `packages` as in the main bridge.

File: src/bridge/bridge.c [hunk 3]
```diff
@@ -574,6 +586,24 @@ run_bridge (const gchar *interactive,
...
+
+  router = setup_router (transport, opt_privileged, &bridges);
+
+  cockpit_router_dump_rules (router);
+
+  g_object_unref (router);
+  g_list_free (bridges);
+  g_object_unref (transport);
+  cockpit_packages_free (packages);
+  packages = NULL;
```
Comment (martinpitt) [inline src/bridge/bridge.c]: Seems a bit odd to free a global variable here, the function neither initializes nor touches that in any way. I see that it just cleans up after that side effect of `setup_router()`. Of course, feel free to ignore this if setting it up in a "nice" way (e. g. making packages an argument of `setup_router()` and keeping it local) is too cumbersome.

File: src/bridge/bridge.c [hunk 4]
```diff
@@ -416,14 +416,24 @@ setup_router (CockpitTransport *transport,
...
   add_router_channels (router);
 
+  /* Enumerated in reverse, since the last rule is matched first.
+   * This has to happen after add_router_channels as the
+   * packages based bridges should have priority.
+   */
+  for (l = g_list_last (bridges); l != NULL; l = g_list_previous (l))
+    {
+      cockpit_router_add_bridge (router, l->data);
+    }
```
Comment (martinpitt) [inline src/bridge/bridge.c]: TBH it's not really clear to me why the list has to be reversed -- is the *intent* that the last rule is matched first? Or is that reversal the workaround for that, to provide a "first match wins" semantics?

Also, this is just a single-line block, so the { } could be dropped.

Comment (petervo) [inline src/bridge/bridge.c]: Last rule added is matched against first but we want the bridges ordered by priority so we add them in reverse. This is all spelled out in cockpit router. The important change here is that add_router_channels is added before the bridges instead of after as was happening before.

File: src/bridge/bridge.c (hunk 3 above)
Comment (petervo) [inline src/bridge/bridge.c]: I don't want to get into a bunch of refactoring here, packages is used in several places. Marius is likely to need to changes this all soon anyways.

I can move the free to the end of the file if that makes things better for now.

File: src/bridge/stub.c (hunk 1 above)
Comment (petervo) [inline src/bridge/stub.c]: moved commit

File: src/bridge/bridge.c (hunk 4 above)
Comment (martinpitt) [inline src/bridge/bridge.c]: Oh, I see that `cockpit_router_new()` does the same reverse-order addition, so it's the same behaviour. That still doesn't explain the "why", as neither doc/guide/packages.xml nor cockpitrouter.c actually define which match wins (first or last) if they have the same priority; so it would be nice to improve the comment here and add it to the docs. That doesn't need to happen in this PR, though.

File: src/bridge/bridge.c [hunk 5]
```diff
@@ -62,7 +62,7 @@
    of the user that is logged into the Server Console.
 */
 
-static CockpitPackages *packages;
+static CockpitPackages *packages = NULL;
```
Comment (martinpitt) [inline src/bridge/bridge.c]: For the record: This is unnecessary (but harmless), static variables are always initialized to 0.
